### ISSUE TITLE
Get output after forgotten closing $ in \todo

### ISSUE THREAD

**Issue by Friedrich2** [henrikmidtiby/todonotes#62]
First, thanks for your great work!

When using the MWE below, one finds that the commented lines make XeLaTeX not produce any output. Instead, one comes into an endless loop, which can be ended by providing \end, but still no page of output is written.

For a simple and often-occurring error like a forgotten closing $, this behaviour does not seem to be appropriate to me. I would be happy, if instead of this behaviour an error message was printed, maybe a visually devastated \todo too, but if the following LaTeX source code (beginning with the next paragraph) was typeset correctly and an output PDF file was produced. This leads me to ask: Is it possible to catch forgotten (closing) $ signs earlier than in the current version?
As an inspiration, in normal text mode followed by inline math, even a forgotten closing $ does only influence the look up to the paragraph end. (c.f. the first line within the MWE result)

My MWE:
```
\documentclass{article}
\usepackage{todonotes}
\begin{document}
Test. $\lambda sdfwldoeäöl \\

Beispieltext\todo{hallo $\lambda$ beispiel} \\ % o. k. Interestingly, the second row seems to be a little bit capped from above, but this is not relevant here.
Beispieltext2\todo{hallo \lambda$ beispiel2} \\ % two errors, but finally compiling
%Beispieltext3\todo{hallo $\lambda beispiel3} \\ % no pages of output, if the user gave ENTER, s, \end and similar. (at least in the way how I did it.)
%Beispieltext4\todo{hallo \lambda beispiel4} \\\  % no pages of output as well
Test line.
\end{document}
```
[20221122_Experiment todonotes.pdf](https://github.com/henrikmidtiby/todonotes/files/10069747/20221122_Experiment.todonotes.pdf)

Note that the commented lines need to be uncommented to reproduce all of the watched behaviour.

**Comment by FrankMittelbach** [comment]
Am 22.11.22 um 18:35 schrieb Friedrich2:
> For a simple and often-occurring error like a forgotten closing $, this 
> behaviour does not seem to be appropriate to me. I would be happy, if 
> instead of this behaviour an error message was printed, maybe a visually 
> devastated \todo too, but if the following LaTeX source code (beginning 
> with the next paragraph) was typeset correctly and an output PDF file 
> was produced. This leads me to ask: Is it possible to catch forgotten 
> (closing) $ signs earlier than in the current version?
> As an inspiration, in normal text mode followed by inline math, even a 
> forgotten closing $ does only influence the look up to the paragraph 
> end. (c.f. the first line within the MWE result)

in my opinion you ask the impossible (in practical terms). Detection of 
a missing $ (either end) happens at fairly random places and is very 
low-level TeX. How much harm that does depends on when TeX detects it. 
However, one gets a useful error message even if the rest of the run may 
be messed up, that is basically the best you can hope for.

It is simply not feasible for \todo to analyze its argument, it is 
simply passed along to other commands, e.g., tikz and then  the error 
happens when that inner code actually tries to typeset it. Once that 
error happens, TeX (not LaTeX) does its own error recovery and that 
sometimes works and sometimes doesn't.

Basically, if you get low-level TeX errors then hitting RETURN will 
seldom keep you going --- and there is not much developers can do about 
it, because most of the time you cannot "detect" the issue without 
producing it (or without implementing TeX in TeX macros ... which is 
possible (aka Turing complete) but not feasible (not with today's computers)
